MapEditor/Components/MovementHandler.cs

```diff
@@ -0,0 +1,269 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using MapEditor.Common;
+using MapEditor.Entities;
+
+namespace MapEditor.Components
+{
+    // https://gamedevelopment.tutsplus.com/series/understanding-steering-behaviors--gamedev-12732
+    public enum MovementMode
+    {
+        Move = 0,
+        Follow,
+        Intercept,
+        Evade,
+        Roam,
+        Patrol,
+    }
+
+    public class MovementHandler : IComponentHandler
+    {
+        // todo: ensure this is checked on loan, not during game runtime
+        public IEnumerable<Type> RequiredComponents { get; } = new[]
+        {
+            typeof(PositionComponent),
+            typeof(MovementComponent),
+            typeof(PhysicsComponent),
+            typeof(PathingComponent)
+        };
+
+        private readonly CollisionController _collisions;
+
+        public MovementHandler(CollisionController collisionController)
+        {
+            _collisions = collisionController;
+        }
+
+        public void Handle(Entity entity, double elapsed)
+        {
+            //todo: incorporate elapsed some how
+
+            // todo: make a GetComponents() method
+            var positionData = entity.GetComponent<PositionComponent>();
+            var movementData = entity.GetComponent<MovementComponent>();
+            var physicsData = entity.GetComponent<PhysicsComponent>();
+            var pathingData = entity.GetComponent<PathingComponent>();
+
+            var destinations = pathingData.Destinations;
+            var position = positionData.Position;
+            var velocity = movementData.Velocity;
+            var slowRadius = movementData.SlowRadius;
+            var stopRadius = movementData.StopRadius;
+            var maxVelocity = movementData.MaxVelocity;
+            var force = movementData.Force;
+            var mass = physicsData.Mass;
+
+            ITarget target;
+            Vector2 steering = null;    //todo: save previous steering, then do steering += Move()
+            switch (movementData.MovementMode)
```

Comment:
Very long class - should move switch into seperate method? of make MovementData a class and have it implement its own movement?